### PR TITLE
Assets cleanups

### LINKED ISSUE #28: "pieces delete" would be nice to show the contents of the snippet before deleting
currently when I delete a snippet it just shows this

Screenshot 2024-02-08 at 12 47 08 PM

it would be nice to show more information, so i can confirm the snippet that i am deleting

### LINKED ISSUE #56: Open command is not working
image

It should open the second asset instead of listing the assets

### PR DESCRIPTION
**Pull request by BishoyHanyRaafat**
1. change the open snippets dir to be the current dir that the user is writing the command
2. split the command_functions.py #54
3. split the list_command #66 
4. fix #56 
5. fix #28 #69

### CODE REVIEW

**Review by hal-8999-alpha — COMMENTED** [review]
Code looks good overall, I ran it locally and I'm still getting index out of range. Specifically, from commands.py. I think this is a caching issue on my end (Yes I used a different env). I checked the error below and it pertains to how the code was previously written. I'm going to try to isolate all my pieces files and delete them but in the meantime it would be good to get someone else to make sure they can run it. 

AppData\Roaming\Python\Python310\site-packages\pieces\commands.py", line 142, in list_assets first_id = ids[0]

**Comment by BishoyHanyRaafat** [comment]
Can u provide more details like a screenshot

**Comment by BishoyHanyRaafat** [comment]
And wait there is not file called commands.py

File: src/pieces/gui.py [hunk 1]
```diff
@@ -78,10 +78,10 @@ def print_help():
...
     print("  run             - Starts a looped version of the CLI that only requires you to type the flag")
     print()
-    print("  list            - List all assets")
-    print("  list x          - Lists maximum of x number of assets")
-    print("  list apps       - List all registered applications")
-    print("  list models     - List all registered AI models")
+    print("  list            - List all assets defaults to 10 assets")
+    print("  list_assets x   - Lists maximum of x number of assets")
+    print("  list_apps       - List all registered applications")
+    print("  list_models     - List all registered AI models")
```
Comment (mark-at-pieces) [inline src/pieces/gui.py:84]: I would still like the commands to be "list assets" and "list apps" and "list models",

so lets move those back to how those were before but good job splitting up the code into smaller more digestible chunks.

I would also like to see the tests get updated/created for these commands to reflect the changes that you made in here.(this should also be a trend w/ each pr moving forward). IE will want a test with each feature/major adjustment to the code if there is not already one.

**Review by mark-at-pieces — CHANGES_REQUESTED** [review]
couple things:
1. lets keep the commands how they are("list assets",...etc), but really good work splitting up the code
2. lets add/adjust the existing tests

**Comment by BishoyHanyRaafat** [comment]
@mark-at-pieces

File: src/pieces/commands/assets.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+def list_command(**kwargs):
+    type = kwargs.get("type","assets")
+    max_assets = kwargs.get("max_assets",10)
+    if type == "assets":
+        list_assets(max_assets)
+    elif type == "apps":
+        list_apps()
+    elif type == "models":
+        list_models()
```
Comment (mark-at-pieces) [inline src/pieces/commands/assets.py:19]: solid

**Review by mark-at-pieces — APPROVED** [review]
LGTM

**Pull request merged**